WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteViewModel.kt

```diff
@@ -169,9 +168,10 @@ class MySiteViewModel @Inject constructor(
             quickStartCategories,
             pinnedDynamicCard,
             visibleDynamicCards,
-            cards
+            cardsUpdate
     ) ->
         val state = if (site != null) {
+            cardsUpdate?.showSnackbarError?.takeIf { it }?.let { cardsUpdate.handleCardsUpdateForSnackbarError() }
```

Comment:
1. Question (❓): It wasn't possible to somehow avoid doing that and instead moving that logic to the `MySiteUiState.update(...)` function as we were discussing, was it? I mean, to avoid having a `var` `showSnackbarError` data class field and then setting it to `false` after the snackbar error is shown.
2. Suggestion (💡): If the above is not possible, consider moving this logic to `MySiteUiState` like below:

`MySiteUiState`
```
data class CardsUpdate(...) : PartialState() {
    fun checkAndShowSnackbarErrorOnce(action: () -> Unit) {
        if (showSnackbarError) {
            action()
            showSnackbarError = false
       }
    }
}
```

Then, its usage can end up being as simple as the below:

`MySiteViewModel`
```
cardsUpdate?.checkAndShowSnackbarErrorOnce {
    _onSnackbarMessage.postValue(Event(SnackbarMessageHolder(UiStringRes(string.my_site_dashboard_update_error))))
}
```

Reply:
1. I wasn't able to make it work until now. 🙂  Can you check [this solution](https://github.com/wordpress-mobile/WordPress-Android/compare/issue/issue/15625-page-behaviours-error-state-part-ii...snackbar-display-logic-update?expand=1)? I can merge the branch so that you can review it.

2. Nice solution! Let me hear from you on 1. again before I make any changes. Maybe we can sync on Slack.

Reply:
👋 @ashiagr !

Aw, I actually like [this solution](https://github.com/wordpress-mobile/WordPress-Android/compare/issue/issue/15625-page-behaviours-error-state-part-ii...snackbar-display-logic-update?expand=1) you are suggesting more. So, to summarize what it does is checking whether this is a non `CardsUpdate` partial state, and if so, it sets the `showSnackbarError` to `false` every time, no matter if that was `true` or `false` before. And this logic is ignored when the partial state is a `CardsUpdate` to make sure that if the `showSnackbarError` is passed as `true` then that is not ignored. I think I like it and kudos for coming up with it. 🥇

Reply:
Yay! I'll merge the branch now and make this PR ready for review so you can finalize your feedback. Thank you for discussing it out loud, that helped. 🙇‍♀️